libraries/sync/src/main/java/taiwan/no/one/sync/data/remote/services/firebase/v1/FirebaseSyncService.kt

```diff
@@ -111,30 +119,32 @@
         }
     }
 
-    override suspend fun modifySong() = TODO()
-
     override suspend fun createSong(song: SimpleTrackEntity) = suspendCancellableCoroutine<String> { continuation ->
+        // Create a document of a song.
         val doc = firestore.collection(COLLECTION_SONG).document(song.obtainTrackAndArtistName())
+        // Set the field detail.
         doc.set(song.toSet())
             .addOnSuccessListener { continuation.resume(doc.path) }
             .addOnFailureListener(continuation::resumeWithException)
     }
 
-    override suspend fun createPlaylistRefToAccount(userInfo: UserInfoEntity, refPlaylistPath: String) =
+    override suspend fun createPlaylistRefToAccount(userInfo: UserInfoEntity, refPlaylistPaths: List<String>) =
         suspendCancellableCoroutine<Boolean> { continuation ->
+            val paths = refPlaylistPaths.map(firestore::document).toTypedArray()
             getUserInfoDocument(userInfo)
-                .update(FIELD_PLAYLIST, FieldValue.arrayUnion(firestore.document(refPlaylistPath)))
+                .update(FIELD_PLAYLIST, FieldValue.arrayUnion(*paths))
```

Comment:
> __[detekt]__ In most cases using a spread operator causes a full copy of the array to be created before calling a method. This may result in a performance penalty. ([view](https://sider.review/gh/repos/263073196/pulls/37#issue-5646162))
>
> <details>
>
> | Rule | Severity |
> | ---- | -------- |
> | `detekt.SpreadOperator` | `warning` |
>
> You can [close this issue](https://sider.review/gh/repos/263073196/pulls/37#issue-5646162) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

It's fine for now